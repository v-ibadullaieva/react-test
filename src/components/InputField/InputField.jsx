import React from "react";
import PropTypes from "prop-types";
import { Input, FormFeedback } from "reactstrap";

const InputField = ({
  input,
  type,
  name,
  placeholder,
  meta: { touched, error }
}) => (
  <div>
    <Input
      {...input}
      type={type}
      name={name}
      placeholder={placeholder}
      invalid={!!(touched && error)}
    />
    <FormFeedback>{error}</FormFeedback>
  </div>
);

InputField.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool.isRequired,
    error: PropTypes.string
  }).isRequired
}

export default InputField;
