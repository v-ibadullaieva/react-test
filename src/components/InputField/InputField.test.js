import React from "react";
import renderer from "react-test-renderer";
import InputField from "./InputField";

const props = {
  input: { value: "hello" },
  meta: { touched: true },
  type: "number",
  name: "name",
  placeholder: "hello"
};

it("renders without crashing", () => {
  const tree = renderer.create(<InputField {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
