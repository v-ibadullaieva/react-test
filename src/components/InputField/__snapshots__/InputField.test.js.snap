// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <input
    className="form-control"
    name="name"
    placeholder="hello"
    type="number"
    value="hello"
  />
  <div
    className="invalid-feedback"
  />
</div>
`;
